gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -122,6 +122,19 @@ public TransportChannelProvider withEndpoint(String endpoint) {
     return toBuilder().setEndpoint(endpoint).build();
   }
 
+  @Override
+  @BetaApi("The surface for customizing pool size is not stable yet and may change in the future.")
+  public boolean acceptsPoolSize() {
+    return poolSize == null;
+  }
+
+  @Override
+  @BetaApi("The surface for customizing pool size is not stable yet and may change in the future.")
+  public InstantiatingGrpcChannelProvider withPoolSize(int size) {
+    Preconditions.checkState(acceptsPoolSize(), "pool size already set to %s", poolSize);
```

Comment:
The way this interface works, if a poolsize is already configured then `provider.withPoolSize(n)` fails, but `provider.toBuilder().setPoolSize(n).build()` succeeds. Could this be a problem?

I don't think it's the end of the world since `TransportChannelProvider` interface itself doesn't have `toBuilder` method.

Reply:
Well it's fine to start with this because it is the more restrictive way of working; we can relax the restriction later.